codalab-service.py

```diff
@@ -0,0 +1,387 @@
+
+        stop_cmd = subparsers.add_parser('stop', help='Stop any existing CodaLab backend instance')
+        down_cmd = subparsers.add_parser('down', help='Bring down any existing CodaLab backend instance')
+        restart_cmd = subparsers.add_parser('restart', help='Restart any existing CodaLab backend instance')
+
+        #  BUILD SETTINGS
+
+        for cmd in [build_cmd, start_cmd]:
+            cmd.add_argument('--version', '-v', type=str, help='CodaLab version to use for building and deployment', default=argparse.SUPPRESS)
+            cmd.add_argument('--dev', action='store_true', help='If specified use dev versions of images', default=argparse.SUPPRESS)
+            cmd.add_argument('--push', action='store_true', help='If specified push the images to Dockerhub', default=argparse.SUPPRESS)
+            cmd.add_argument('--docker-user', type=str, help='DockerHub username to push images from', default=argparse.SUPPRESS)
+            cmd.add_argument('--docker-pwd', type=str, help='DockerHub password to push images from', default=argparse.SUPPRESS)
+
+        build_cmd.add_argument('image', default=argparse.SUPPRESS, help='Images to build', choices=['bundleserver', 'frontend', 'worker', 'default-cpu', 'default-gpu', 'all'], nargs='?')
+
+        #  DEPLOYMENT SETTINGS
+        for cmd in [start_cmd, stop_cmd, restart_cmd, down_cmd, logs_cmd]:
+            cmd.add_argument('--instance-name', type=str, help='Docker compose name to use for instance', default=argparse.SUPPRESS)
+
+        start_cmd.add_argument('--build-locally', '-b', action='store_true', help='If specified build VERSION using local code.', default=argparse.SUPPRESS)
+        start_cmd.add_argument('--test-build', '-t', action='store_true', help='If specified run tests on the build.', default=argparse.SUPPRESS)
+        start_cmd.add_argument('--user-compose-file', type=str, help='If specified path to a user-defined Docker compose file that overwrites the defaults', default=argparse.SUPPRESS)
+        start_cmd.add_argument('--start-worker', '-w', action='store_true', help='If specified start a CodaLab worker on this machine.', default=argparse.SUPPRESS)
+        start_cmd.add_argument('--initial-config', '-i', action='store_true', help='If specified, save the initial configuration of the instance (defaults to true if the service home or the database mounts are ephemeral)', default=argparse.SUPPRESS)
+
+        #  USER CREDENTIALS
+
+        start_cmd.add_argument('--mysql-root-pwd', type=str, help='Root password for the database', default=argparse.SUPPRESS)
+        start_cmd.add_argument('--mysql-user', type=str, help='MYSQL username for the Codalab MYSQL client', default=argparse.SUPPRESS)
+        start_cmd.add_argument('--mysql-pwd', type=str, help='MYSQL password for the Codalab MYSQL client', default=argparse.SUPPRESS)
+        start_cmd.add_argument('--root-user', type=str, help='Codalab username for the Codalab admin user', default=argparse.SUPPRESS)
```

Comment:
maybe this should be `--codalab-user` (root is vague in the context of these other options)

Reply:
Changed